### DISCUSSION TITLE
What is nuclei-4234140034? I did not perform any tasks

### DISCUSSION THREAD

**Discussion by hktalent** [projectdiscovery/nuclei#1900] [Q&A] [answered]
what this?
can not stop
can not
rm -rf /Users/51pwn/.cache
?
why?

Chromium  49943 51pwn   19u  IPv4 0x5e64bd050af8db9b      0t0  TCP 127.0.0.1:64276 (LISTEN)
what is:
"--user-data-dir=/var/folders/_l/pnb2t_9s0f192bqlz1348vpr0000gn/T/nuclei-4234140034"
in under:

/Users/51pwn/.cache/rod/browser/chromium-991974/chrome-mac/Chromium.app/Contents/MacOS/Chromium --ignore-certificate-errors=1 --disable-renderer-backgrounding --mute-audio=true --no-first-run --disable-popup-blocking --force-color-profile=srgb --disable-gpu=true --window-size=1080,1920 --headless --no-startup-window --disable-background-networking --disable-ipc-flooding-protection --disable-prompt-on-repost --disable-sync --disable-default-apps --disable-hang-monitor --user-data-dir=/var/folders/_l/pnb2t_9s0f192bqlz1348vpr0000gn/T/nuclei-4234140034 --disable-features=site-per-process,TranslateUI --hide-scrollbars=true --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-component-extensions-with-background-pages --disable-dev-shm-usage --disable-notifications=true --disable-breakpad --disable-client-side-phishing-detection --remote-debugging-port=0 --disable-blink-features=AutomationControlled --incognito=true --enable-automation --metrics-recording-only --enable-features=NetworkService,NetworkServiceInProcess --disable-crash-reporter=true

**Comment by forgedhallpass** [comment, marked as answer]
Headless templates are executed through a Chromium process in the background. The browser interactions are achieved through the [go-rod](https://github.com/go-rod/rod) library. That is what you see in the process list, running from the default template directory. In order to run these templates, you have to explicitly set the `-headless` flag. If you would like to see the actions performed visually on the browser, you can use the `-show-browser` flag.

For more information please visit: https://nuclei.projectdiscovery.io/templating-guide/protocols/headless/

**Comment by hktalent** [comment]
@forgedhallpass I know, but I've stopped running nucleus for a long time. This process has been running all the time, and it also listens to the port, which makes me nervous

**Reply by forgedhallpass** [reply]
It depends on how nuclei was stopped. We have logic that stops chromium processes opened by nuclei (see https://github.com/projectdiscovery/nuclei/blob/dc46bd263b97de6d0fa50d2e7829e718ed366289/v2/pkg/protocols/headless/engine/engine.go#L134), but in certain cases those processes might remain hanging. If you see this happening often after nuclei has closed gracefully and you can reproduce this behavior reliably, please create an issue with all the possible details and steps, and we'll look into it. I'd like to mention again, that chromium is only used by templates requiring browser interactions, and they need to be enabled explicitly through the `-headless` flag.

p.s. Both `nuclei`, `go-rod` and `chromium` are an open-source projects. You are invited to analyze the source code and provide your feedback publicly.
